src/concurrency/init_once.rs

```diff
@@ -172,7 +173,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
 
         // Each complete happens-before the end of the wait
         if let Some(data_race) = &this.machine.data_race {
-            data_race.validate_lock_release(&mut init_once.data_race, current_thread);
+            data_race.validate_lock_release(&mut init_once.data_race, current_thread, current_span);
```

Comment:
Would it make sense to move the `current_span` calls behind the fn call? That seems nicer than putting the burden on all callers.  Basically instead of passing current_thread and current_span it could pass `&Machine`?

EDIT: Ah, that doesn't work since it overlaps with `data_race`. Hm... I guess we can go ahead with separate arguments for now, but if you have a nice idea for how to avoid having to pass so many arguments, that would be great.